src/main/plugins/org.dita.base/xsl/common/functions.xsl

```diff
@@ -48,9 +48,12 @@ See the accompanying LICENSE file for applicable license.
   <xsl:function name="dita-ot:retrieve-href-target" as="node()?">
     <xsl:param name="href" as="attribute(href)"/>
 
-    <xsl:variable name="doc" as="document-node()"
```

Comment:
Because [`doc` will return an empty node sequence when the resource cannot be read](https://www.w3.org/TR/xquery-operators/#func-doc), can't we just change the type to `document-node)? `. Then we would not have to use `doc-available`.

Reply:
That's what I started with - but it still crashed. Just running `doc(dita-ot:resolve-href-path($href))` in the `select` attribute crashed the build when the document was not available.

Reply:
Ah, sorry. I got confused. You should be using [`document()` function from XSLT](https://www.w3.org/TR/xslt20/#document), not `doc()` from XPath. The former can handle missing document, the latter throws an error.

Reply:
Ah, yeah, changing to `document-node()?` and to `document()` also fixed it. On the other hand, using `document()` also added a lot more XSLT warnings -- so using `doc-available` might be better? The `document()` approach gives warnings like this (and we've already warned earlier in the build about the missing file):

```
     [xslt] Warning at char 35 in xsl:variable/@select on line 53 column 60 of functions.xsl:
     [xslt]   FODC0002: Document has been marked not available:
     [xslt]   file:/Users/robander/Documents/github/GITHUBTESTS/term/temp/missing.dita
```